daemon/smartcard.c

```diff
@@ -158,7 +268,8 @@ smartcard_cb_start_container(int fd, unsigned events, event_io_t *io, void *data
 				      container_get_name(startdata->container), keyfile);
 				unsigned char key[TOKEN_MAX_WRAPPED_KEY_LEN];
 				int keylen = file_read(keyfile, (char *)key, sizeof(key));
-				if (keylen != sizeof(key)) {
+				DEBUG("Length of existing key: %d", keylen);
+				if (keylen < 0) {
```

Comment:
if the key was wrapped with the symmetric key of the usb token, the "envelope" is just a few bytes larger than the key itself. the previous check if the keyfile has been read correctly does not work anymore in that case